publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/PublicationPermissionStrategyTest.java

```diff
@@ -45,170 +58,256 @@ class PublicationPermissionStrategyTest {
+    private static final String EXTERNAL_ISSUER = ENVIRONMENT.readEnv("EXTERNAL_USER_POOL_URI");
+    private static final String EXTERNAL_CLIENT_ID = "external-client-id";
+
+    private static final URI EXTERNAL_CLIENT_CUSTOMER_URI = URI.create("https://example.com/external-client-org");
+
+    @BeforeEach
+    void setUp() throws NotFoundException {
+        this.identityServiceClient = mock(IdentityServiceClient.class);
+        when(this.identityServiceClient.getExternalClient(any())).thenReturn(
+            new GetExternalClientResponse(randomString(), randomString(), EXTERNAL_CLIENT_CUSTOMER_URI, randomUri()));
+    }
 
     @Test
-    void shouldDenyPermissionToDeletePublicationWhenUserHasNoAccessRights() throws JsonProcessingException {
-        var requestInfo = createRequestInfo(randomString(), randomUri());
+    void shouldDenyPermissionToUnpublishPublicationWhenUserHasNoAccessRights()
+        throws JsonProcessingException, UnauthorizedException {
+        var cristinId = randomUri();
+        var requestInfo = createRequestInfo(randomString(), randomUri(), new ArrayList<>(), cristinId);
         var publication = createPublication(randomString(), randomUri());
 
         Assertions.assertFalse(PublicationPermissionStrategy
-                                   .fromRequestInfo(requestInfo)
-                                   .hasPermissionToUnpublish(publication));
+                                   .create(publication, RequestUtil.createUserInstanceFromRequest(
+                                                        requestInfo, identityServiceClient))
+                                   .hasPermission(PublicationAction.UNPUBLISH));
     }
 
     @Test
-    void shouldDenyPermissionToDeletePublicationWhenUserMissingBasicInfo() throws JsonProcessingException {
-        var requestInfo = createRequestInfo(null, URI.create(""));
+    void shouldDenyPermissionToUnpublishPublicationWhenUserMissingBasicInfo()
+        throws JsonProcessingException {
+        var cristinId = randomUri();
+        var requestInfo = createRequestInfo(null, URI.create(""), cristinId);
         var publication = createPublication(randomString(), randomUri());
 
-        Assertions.assertFalse(PublicationPermissionStrategy
-                                   .fromRequestInfo(requestInfo)
-                                   .hasPermissionToUnpublish(publication));
+        Assertions.assertThrows(UnauthorizedException.class, () -> PublicationPermissionStrategy
+                                   .create(publication, RequestUtil.createUserInstanceFromRequest(
+                                                        requestInfo, identityServiceClient))
+                                   .hasPermission(PublicationAction.UNPUBLISH));
     }
 
     @Test
-    void shouldGiveEditorPermissionToDeletePublicationWhenPublicationIsFromTheirInstitution()
-        throws JsonProcessingException {
+    void whenDegreeAndDraftStatusAllowUpdateForResourceOwner()
```

Comment:
shouldReturnTrueWhenSomething, but I'm not sure what something is right now…possibly EditorUpdatesDegreeFromOwnInstitution?

Reply:
I made this to get test coverage, but this test is both weird I see now and it's not connected to the left side here at all. I changed the action to UPDATE to make a bit more sense and gave it a new name shouldAllowResourceOwnerToUpdateDegreeInDraftStatus.